Naming/SubdirDirectoryPropertyNamer.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Vich\UploaderBundle\Naming;
+
+use Vich\UploaderBundle\Mapping\PropertyMapping;
+use Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException;
+use Vich\UploaderBundle\Exception\NameGenerationException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * Directory namer which can create subfolder depends on property.
+ *
+ * @author Raynald Coupé <raynald@easi-services.fr>
+ */
+class SubdirDirectoryPropertyNamer implements DirectoryNamerInterface, ConfigurableInterface
+{
+    /**
+     * @var string
+     */
+    private $propertyPath;
+
+    /**
+     * @param array $options Options for this namer. The following options are accepted:
+     *                       - property: path to the property used to name the file. Can be either an attribute or a method.
+     */
+    public function configure(array $options)
+    {
+        if (empty($options['property'])) {
+            throw new \InvalidArgumentException('Option "property" is missing or empty.');
+        }
+
+        $this->propertyPath = $options['property'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function directoryName($object, PropertyMapping $mapping)
+    {
+        if (empty($this->propertyPath)) {
+            throw new \LogicException('The property to use can not be determined. Did you call the configure() method?');
+        }
+
+        try {
+            $name = $this->getPropertyValue($object, $this->propertyPath);
+        } catch (NoSuchPropertyException $e) {
+            throw new NameGenerationException(sprintf('Directory name could not be generated: property %s does not exist.', $this->propertyPath), $e->getCode(), $e);
+        }
+
+        if (empty($name)) {
+            throw new NameGenerationException(sprintf('Directory name could not be generated: property %s is empty.', $this->propertyPath));
+        }
+
+        return $name;
+    }
+
+    private function getPropertyValue($object, $propertyPath)
```

Comment:
I tried to use the **readProperty** method through the **$mapping** parameter of the **directoryName** method but it raise an error because of the **array_key_exists** check
https://github.com/dustin10/VichUploaderBundle/blob/56c0d3ad6fc7c92bae7b05e46b3b9b5be0ec5ecc/Mapping/PropertyMapping.php#L137-L139
The property/method to access (the one added in the config.yml) is not in the **propertyPaths** array.
Any idea ?